NitroxServer/Communication/NetworkingLayer/NitroxServer.cs

```diff
@@ -29,12 +29,13 @@ public NitroxServer(PacketHandler packetHandler, PlayerManager playerManager, En
 
             portNumber = serverConfig.ServerPort;
             maxConn = serverConfig.MaxConnections;
+            SetupUPNP();
```

Comment:
IO should not be done in constructors. Constructors are expected to be fast and not cause side effects. Consider putting it in start by making it virtual or require implementors of this class to call it if needed.